packages/react-composites/src/composites/CallComposite/components/MediaGallery.tsx

```diff
@@ -60,6 +65,25 @@ export const MediaGallery = (props: MediaGalleryProps): JSX.Element => {
     };
   }, [cameraSwitcherCallback, cameraSwitcherCameras]);
 
+  // Async function to retrieve custom data model information to replace display name in remote tiles with camera on.
+  const fetchAvatarPersonaDataAsync = memoizeFunction(
+    async (participant: VideoGalleryRemoteParticipant): Promise<void> => {
+      if (props.onFetchAvatarPersonaData) {
+        const newParticipantData = await props.onFetchAvatarPersonaData(participant.userId);
+        participant.displayName = newParticipantData.text ? newParticipantData.text : participant.displayName;
```

Comment:
Two things here:
1. This won't trigger a re-render to update the display name. It may be working when testing but only because other things will cause rerenders to occur
2. This mutates the participant in `videoGalleryProps.remoteParticipants` which is bad practice. (Mutating variables makes code very difficult to understand and debug as values could have changed randomly at any time, by anyone)

Instead consider having the merged participants in their own state in this component that can trigger re-renders:
```tsx
const [participants, setParticipants] = useState<VideoGalleryRemoteParticipant[]>([]);
```

Then when the participants update, fetch their props asynchronously:
```tsx
useEffect(() => {
  // in async block:
  const augmentedParticipants = await generateVideoParticipantsFromCustomData(videoGalleryProps.remoteParticipants, props.onFetchAvatarPersonaData);
  setParticipants(augmentedParticipants);

  // **note:** This is just a quick example but a poor implementation. If two renders are triggered due to 
  // remoteParticipants changing, there's no guarantee in what order the setParticipants will occur
}, [videoGalleryProps.remoteParticipants]
```